src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
+        }
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size >= objects.length) {
+            this.grow();
+        }
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index: " + index + "size: " + size);
+        }
+        if (index <= size) {
+            Object[] newObjectsArray = new Object[objects.length + 1];
+            System.arraycopy(objects, 0, newObjectsArray, 0, index);
+            newObjectsArray[index] = value;
+            System.arraycopy(objects, index, newObjectsArray, index + 1, objects.length - index);
+            objects = newObjectsArray;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        Object[] newObjectsArray;
+        if (list != null) {
+            int additionalSize = list.size();
+            if (additionalSize > 0) {
+                newObjectsArray = new Object[size + additionalSize];
+                System.arraycopy(objects, 0, newObjectsArray, 0, size);
+                for (int i = 0; i < additionalSize; i++) {
+                    newObjectsArray[size + i] = list.get(i);
+                }
+                objects = newObjectsArray;
+            }
+            size += additionalSize;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        return (T) objects[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexCheck(index);
+        objects[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        Object[] newObjectsArray = new Object[objects.length];
+        System.arraycopy(objects, 0, newObjectsArray, 0, index);
+        System.arraycopy(objects, index + 1, newObjectsArray, index, objects.length - (index + 1));
+        T removedObject = (T) objects[index];
+        objects = newObjectsArray;
+        size--;
+        return removedObject;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = -1;
+        T removedObject;
+        Object[] newObjectsArray;
+        for (int i = 0; i < size; i++) {
+            if (element == objects[i]
+                    || (objects[i] != null && objects[i].equals(element))) {
+                index = i;
+            }
+        }
+        if (index == -1) {
+            throw new NoSuchElementException("Element not finded!");
+        } else {
+            removedObject = (T) objects[index];
+            newObjectsArray = new Object[objects.length];
+            System.arraycopy(objects, 0, newObjectsArray, 0, index);
+            System.arraycopy(objects, index + 1, newObjectsArray, index,
+                    objects.length - (index + 1));
+            objects = newObjectsArray;
+            size--;
+        }
+        return removedObject;
```

Comment:
lets just find index of incoming `element` and call `remove(int index)` method

Reply:
Done